src/wp-includes/blocks.php

```diff
@@ -797,34 +802,48 @@ function make_before_block_visitor( $context ) {
 
 		if ( $parent && ! $prev ) {
 			// Candidate for first-child insertion.
-			$hooked_blocks_for_parent = get_hooked_blocks( $parent['blockName'] );
-			foreach ( $hooked_blocks_for_parent as $hooked_block_type => $relative_position ) {
-				if ( 'first_child' === $relative_position ) {
-					$hooked_block_markup = get_comment_delimited_block_content( $hooked_block_type, array(), '' );
-					/** This filter is documented in wp-includes/blocks.php */
-					$markup .= apply_filters( 'inject_hooked_block_markup', $hooked_block_markup, $hooked_block_type, $relative_position, $parent, $context );
-				}
+			$relative_position  = 'first_child';
+			$anchor_block_type  = $parent['blockName'];
+			$hooked_block_types = array_keys( get_hooked_blocks( $anchor_block_type, $relative_position ) );
+			/**
+			 * Filters the list of hooked block types for a given anchor block type and relative position.
+			 *
+			 * @since 6.4.0
+			 *
+			 * @param string[]                $hooked_block_types  The list of hooked block types.
+			 * @param string                  $relative_position   The relative position of the hooked blocks.
+			 *                                                     Can be one of 'before', 'after', 'first_child', or 'last_child'.
+			 * @param string                  $anchor_block_type   The anchor block type.
+			 * @param WP_Block_Template|array $context             The block template, template part, or pattern that the anchor block belongs to.
+			 */
+			$hooked_block_types = apply_filters( 'hooked_block_types', $hooked_block_types, $relative_position, $anchor_block_type, $context );
```

Comment:
Should we call the filter just `hooked_blocks`?

Reply:
I like `hooked_block_types` personally, but defer to you all. It makes it clear to me that you are filtering the "block types" and not the entire block markup or something else.

Reply:
Agree with Nick here although it does make me wonder how consistent the codebase is in differentiating between "blocks" and "block types".

Reply:
> how consistent the codebase is in differentiating between "blocks" and "block types".

😬 I support any effort to draw those distinctions where they haven't already been blurred.

Reply:
`hooked_block_types` sounds good 👍🏻